fdbclient/AuditUtils.actor.cpp

```diff
@@ -0,0 +1,175 @@
+			tr.set(auditKey(auditState.getType(), auditId), auditStorageStateValue(auditState));
+			wait(tr.commit());
+			TraceEvent("PersistedNewAuditState", auditId).detail("AuditKey", auditKey(auditState.getType(), auditId));
+			break;
+		} catch (Error& e) {
+			wait(tr.onError(e));
+		}
+	}
+
+	return auditId;
+}
+
+ACTOR Future<Void> persistAuditState(Database cx, AuditStorageState auditState) {
+	state Transaction tr(cx);
+
+	loop {
+		try {
+			tr.set(auditKey(auditState.getType(), auditState.id), auditStorageStateValue(auditState));
+			wait(tr.commit());
+			break;
+		} catch (Error& e) {
+			wait(tr.onError(e));
+		}
+	}
+
+	return Void();
+}
+
+ACTOR Future<AuditStorageState> getAuditState(Database cx, AuditType type, UID id) {
+	state Transaction tr(cx);
+	state Optional<Value> res;
+
+	loop {
+		try {
+			Optional<Value> res_ = wait(tr.get(auditKey(type, id)));
+			res = res_;
+			TraceEvent("ReadAuditState", id).detail("AuditKey", auditKey(type, id));
+			break;
+		} catch (Error& e) {
+			wait(tr.onError(e));
+		}
+	}
+
+	if (!res.present()) {
+		throw key_not_found();
+	}
+
+	return decodeAuditStorageState(res.get());
+}
+
+ACTOR Future<std::vector<AuditStorageState>> getLatestAuditStates(Database cx, AuditType type, int num) {
+	Transaction tr(cx);
+	std::vector<AuditStorageState> auditStates = wait(getLatestAuditStatesImpl(&tr, type, num));
+	return auditStates;
+}
+
+ACTOR Future<Void> persistAuditStateMap(Database cx, AuditStorageState auditState) {
+	state Transaction tr(cx);
+
+	loop {
+		try {
+			wait(krmSetRange(
+			    &tr, auditRangePrefixFor(auditState.id), auditState.range, auditStorageStateValue(auditState)));
+			break;
+		} catch (Error& e) {
+			wait(tr.onError(e));
+		}
+	}
+
+	return Void();
+}
+
+ACTOR Future<std::vector<AuditStorageState>> getAuditStateForRange(Database cx, UID id, KeyRange range) {
+	state RangeResult auditStates;
+	state Transaction tr(cx);
+
+	loop {
+		try {
+			RangeResult res_ = wait(krmGetRanges(&tr,
+			                                     auditRangePrefixFor(id),
+			                                     range,
+			                                     CLIENT_KNOBS->KRM_GET_RANGE_LIMIT,
+			                                     CLIENT_KNOBS->KRM_GET_RANGE_LIMIT_BYTES));
+			auditStates = res_;
+			break;
+		} catch (Error& e) {
+			wait(tr.onError(e));
+		}
+	}
+
+	std::vector<AuditStorageState> res;
+	for (int i = 0; i < auditStates.size() - 1; ++i) {
+		KeyRange currentRange = KeyRangeRef(auditStates[i].key, auditStates[i + 1].key);
+		AuditStorageState auditState;
+		if (!auditStates[i].value.empty()) {
+			AuditStorageState auditState = decodeAuditStorageState(auditStates[i].value);
+		}
+		auditState.range = currentRange;
+		res.push_back(auditState);
+	}
+
+	return res;
+}
+
+StringRef auditTypeToString(const AuditType type) {
+	switch (type) {
+	case AuditType::Invalid:
+		return "Invalid"_sr;
+	case AuditType::ValidateHA:
+		return "ValidateHA"_sr;
+	}
+	return "Invalid"_sr;
```

Comment:
Is it intentional that the default case is same as for AuditType::Invalid: ?

Reply:
Yes, I want to make sure the value is set explicitly.